src/lib/monitor.ml

```diff
@@ -0,0 +1,210 @@
+
+type pipeline_tree =
+  | Item : 'a Current.t -> pipeline_tree
+  | Seq of (string * pipeline_tree) list
+  | And of (string * pipeline_tree) list 
+
+type t = {
+  mutable blessing: Package.Blessing.Set.t Current.t OpamPackage.Map.t;
```

Comment:
Why do you need to keep the blessed packages?

Reply:
To explain the failure of a given opam package we need to know which compilation universe is the blessed one to figure out why it failed. For now we don't care about the other compilation universes but they could also be displayed at some point.

Reply:
I see! Thanks for the clarification :)